src/App.php

```diff
@@ -112,6 +112,15 @@ class App
     /** @var array<string, View> Modal view that need to be rendered using JSON output. */
     private $portals = [];
 
+    /**
+     * @var string used in method App::url to build the url
+     *
+     *  Used only in method App::url
+     *  If filename part is missing during building of url, this page will be used
+     *  Hint: if you use a routing system, you need to set this and urlBuildingExt to empty string
+     */
+    protected $urlBuildingPage = 'index';
```

Comment:
Is `index.php` needed by some major webserver? If not, what about dropping it completely?